test/torch/pointers/test_pointer_tensor.py

```diff
@@ -309,3 +310,67 @@ def test_raising_error_when_item_func_called(workers):
     pointer = PointerTensor(id=1000, location=workers["alice"], owner=workers["me"])
     with pytest.raises(RuntimeError):
         pointer.item()
+
+
+def test_fix_prec_on_pointer_tensor(workers):
+    """
+    Ensure .fix_precision() works as expected.
+    """
+    bob = workers["bob"]
+
+    tensor = torch.tensor([1, 2, 3, 4.0])
+    ptr = tensor.send(bob)
+
+    ptr = ptr.fix_precision()
+
+    assert type(ptr.child) is PointerTensor
+    assert type(ptr.get().child) is FixedPrecisionTensor
+
+
+def test_fix_prec_on_pointer_of_pointer(workers):
+    """
+    Ensure .fix_precision() works along a chain of pointers.
+    """
+    bob = workers["bob"]
+    alice = workers["alice"]
+
+    tensor = torch.tensor([1, 2, 3, 4.0])
+    ptr = tensor.send(bob)
+    ptr = ptr.send(alice)
+
+    ptr = ptr.fix_precision()
+
+    assert type(ptr.child) is PointerTensor
+    assert type(ptr.get().get().child) is FixedPrecisionTensor
+
+
+def test_float_prec_on_pointer_tensor(workers):
+    """
+    Ensure .float_precision() works as expected.
+    """
+    bob = workers["bob"]
+
+    tensor = torch.tensor([1, 2, 3, 4.0]).fix_prec()
+    ptr = tensor.send(bob)
+
+    ptr = ptr.float_precision()
+
+    assert type(ptr.child) is PointerTensor
+    assert type(ptr.get()) is torch.Tensor
```

Comment:
This should be a FixedPrecisionTensor, right?

Reply:
- Call fix_precision() on the ptr before calling float_precision() (otherwise it doesn't make sense even if it doesn't fail)
- You can use bob._objects[ptr.id_at_location] to inspect a remote object (if you want to check types for example)
- Prefer maybe isinstance(..., ...) instead of type() == ..., it is more robust for sub classes